packages/protocol/contracts/tokenvault/BridgedERC20.sol

```diff
@@ -33,8 +33,11 @@ contract BridgedERC20 is
     error BTOKEN_CANNOT_RECEIVE();
     error BTOKEN_UNAUTHORIZED();
 
-    modifier onlyOwnerOrSnapshooter() {
-        if (msg.sender != owner() && msg.sender != snapshooter) {
+    modifier onlyAuthorizedForSnapshot() {
+        if (
+            msg.sender != owner() && msg.sender != snapshooter
+                && msg.sender != resolve(LibStrings.B_TAIKO, true)
```

Comment:
So actually `TaikoL2` contract can make a snapshot in any bridged contracts, right ? While indeed it does not doing it automatically just for TKO, right ? Isn't it a risk to any, arbitrary bridged contracts that technically taiko can make a snapshot - and sabotage their DAO ?

Reply:
Maybe a check here like:
```
...
if ( address(this) == resolve(LibStrings.B_TAIKO_TOKEN, false)
{do this check}
else
{do regular check}
...
```